### PR TITLE
Pause before checking session eviction

### PR DESCRIPTION
**Pull request by janbartel**
Tests which check the state of the session cache were checking the cache as soon as the response was received on the client: at this point, the session release handling may not have occurred, and thus it's presence in the cache may not yet have been decided. Fixed by adding a small wait on the client before doing any checking, using `Awaitility` to check the desired cache state.

### CODE REVIEW

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-sessions/jetty-ee10-test-sessions-common/src/test/java/org/eclipse/jetty/ee10/session/AsyncTest.java [hunk 1]
```diff
@@ -130,8 +130,11 @@ public void testSessionWithAsyncComplete() throws Exception
             String id = SessionTestSupport.extractSessionId(sessionCookie);
 
             //session should now be evicted from the cache after request exited
-            assertFalse(contextHandler.getSessionHandler().getSessionCache().contains(id));
             assertTrue(contextHandler.getSessionHandler().getSessionCache().getSessionDataStore().exists(id));
+            Awaitility.await().atMost(30, TimeUnit.SECONDS)
+                .until(() -> !contextHandler.getSessionHandler().getSessionCache().contains(id));
+            assertFalse(contextHandler.getSessionHandler().getSessionCache().contains(id));
```
Comment (olamy) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-sessions/jetty-ee10-test-sessions-common/src/test/java/org/eclipse/jetty/ee10/session/AsyncTest.java:141]: just remove the second `assertFalse` as it's already done by Awaitility

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-sessions/jetty-ee10-test-sessions-common/src/test/java/org/eclipse/jetty/ee10/session/AsyncTest.java [hunk 2]
```diff
@@ -180,6 +183,8 @@ public void testSessionWithCrossContextAsync() throws Exception
 
             //session should now be evicted from the cache after request exited
             String id = SessionTestSupport.extractSessionId(sessionCookie);
+            Awaitility.await().atMost(30, TimeUnit.SECONDS)
+                .until(() -> !contextB.getSessionHandler().getSessionCache().contains(id));
             assertFalse(contextB.getSessionHandler().getSessionCache().contains(id));
```
Comment (olamy) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-sessions/jetty-ee10-test-sessions-common/src/test/java/org/eclipse/jetty/ee10/session/AsyncTest.java:193]: cant be dropped

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-sessions/jetty-ee10-test-sessions-common/src/test/java/org/eclipse/jetty/ee10/session/AsyncTest.java [hunk 3]
```diff
@@ -226,6 +231,8 @@ public void testSessionCreatedBeforeDispatch() throws Exception
             String id = SessionTestSupport.extractSessionId(sessionCookie);
 
             //session should now be evicted from the cache after request exited
+            Awaitility.await().atMost(30, TimeUnit.SECONDS)
+                .until(() -> !contextHandler.getSessionHandler().getSessionCache().contains(id));
             assertFalse(contextHandler.getSessionHandler().getSessionCache().contains(id));
```
Comment (olamy) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-sessions/jetty-ee10-test-sessions-common/src/test/java/org/eclipse/jetty/ee10/session/AsyncTest.java:241]: cant be dropped

**Review by olamy — CHANGES_REQUESTED** [review]
if you do the test using `Awaitility` you can remove the assert* as Awaitility is already the assert

**Review by gregw — APPROVED** [review]

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-sessions/jetty-ee10-test-sessions-common/src/test/java/org/eclipse/jetty/ee10/session/AsyncTest.java (hunk 1 above)
Comment (janbartel) [reply jetty-ee10/jetty-ee10-tests/jetty-ee10-test-sessions/jetty-ee10-test-sessions-common/src/test/java/org/eclipse/jetty/ee10/session/AsyncTest.java:141]: Done. However, my changes seem to not be showing up on github?! See commit `a2c93bdc70`

**Review by olamy — APPROVED** [review]

**Review by mkvl77 — APPROVED** [review]

**Pull request merged**